elastic4s-core/src/main/scala/com/sksamuel/elastic4s/update/UpdateApi.scala

```diff
@@ -10,9 +10,11 @@ trait UpdateApi {
     def in(indexType: IndexAndTypes): UpdateDefinition = UpdateDefinition(indexType, id.toString)
   }
 
-  def updateIn(indexes: Indexes): UpdateExpectsQuery = new UpdateExpectsQuery(indexes)
-  class UpdateExpectsQuery(indexes: Indexes) {
-    def query(query: QueryDefinition) = UpdateByQueryDefinition(indexes, query)
+  def updateIn(indexes: Indexes): UpdateExpectsQuery = new UpdateExpectsQuery(indexes.toIndexesAndTypes)
+  def updateIn(indexesAndTypes: IndexesAndTypes): UpdateExpectsQuery = new UpdateExpectsQuery(indexesAndTypes)
```

Comment:
If you don't think it makes sense to add this in master for 6.x, I'm fine with removing this, but I would like to include it in the 5.5.x branch

Reply:
I think we can add both, but in 6.0 lets immediately deprecate it. This means people who need to update by type can do so, but they'll get the warning that it's going away in 7.